### PR TITLE
added nxos_smoke tests

### PR DESCRIPTION
**Pull request by newswangerd**
##### SUMMARY

covers:
module_utils:
- network.common.utils
- network.common.config
- network.nxos

plugins:
- plugins.action.nxos
- plugins. cliconf.nxos
- plugins.terminal.nxos

Coverage Reports: https://newswangerd.github.io/nxos/

##### ISSUE TYPE

 - test pull request

##### ANSIBLE VERSION

```
ansible 2.6.0 (nxos_smoke b8d9c0b580) last updated 2018/02/23 11:41:13 (GMT -400)
  config file = None
  configured module search path = [u'/Users/david/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/david/code/ansible/lib/ansible
  executable location = /Users/david/code/ansible/bin/ansible
  python version = 2.7.10 (default, Jul 15 2017, 17:16:57) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: test/integration/targets/nxos_smoke/tests/common/common_config.yaml [hunk 1]
```diff
@@ -0,0 +1,158 @@
+---
+# nxos_config -> NetworkConfig, dumps
+# nxos_static_route -> CustomNetworkConfig
+
+# hit NetworkConfig
+# Select interface for test
+- debug: msg="START connection={{ connection }} nxos_smoke common_config test"
+
```
Comment (trishnaguha) [inline test/integration/targets/nxos_smoke/tests/common/common_config.yaml]: Please add this as debug statement too:
```
- debug: msg="Using provider={{ connection.transport }}"
  when: ansible_connection == "local"
```

File: test/integration/targets/nxos_smoke/tests/common/misc_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      - show hostname
+      - show hostname
+      - show hostname
+      - show hostname
+      - show hostname
+      - show hostname
+      - show hostname
+      - show hostname
+      - show hostname
+    provider: "{{ connection }}"
```
Comment (trishnaguha) [inline test/integration/targets/nxos_smoke/tests/common/misc_tests.yaml:19]: `assert` statement seems to be missing here?

File: test/integration/targets/nxos_smoke/tests/common/common_config.yaml [hunk 3]
```diff
@@ -0,0 +1,158 @@
+---
+# nxos_config -> NetworkConfig, dumps
+# nxos_static_route -> CustomNetworkConfig
+
+# hit NetworkConfig
+# Select interface for test
+- debug: msg="START connection={{ connection }} nxos_smoke common_config test"
```
Comment (trishnaguha) [inline test/integration/targets/nxos_smoke/tests/common/common_config.yaml]: How about using this debug statement:
```
- debug: msg="START connection={{ ansible_connection }}/set_hostname.yaml"
- debug: msg="Using provider={{ connection.transport }}"
  when: ansible_connection == "local"
```
This displays which connection plugin it is using and which transport explicitly.

**Review by trishnaguha — APPROVED** [review]

**Review by gundalow — APPROVED** [review]

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Merge and cherry-pick

**Pull request merged**